### PR TITLE
Code clean ups to RulesAttachment. No logic changes.

### PR DESCRIPTION
**Pull request by asvitkine**
## Change Summary & Additional Notes
Fixes 7 code climate issues!

### CODE REVIEW

**Review by RoiEXLab — APPROVED** [review]
It's a bit hard to follow all the individual changes but I think I got it now. Hard to tell if it's completely error-free but I can spot any obvious ones.

Nice change, just a small idea from my side

File: game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java [hunk 1]
```diff
@@ -99,35 +100,28 @@ static RulesAttachment get(
...
-            continue;
-          }
-          ra = (RulesAttachment) otherPlayer.getAttachment(nameOfAttachment);
-          if (ra != null) {
-            return ra;
-          }
+    if (ra != null) {
+      return ra;
+    }
+    if (playersToSearch != null) {
```
[the remark this answers is not in the document]
Comment (asvitkine) [reply game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java]: Done.

[the remark this answers is not in the document]
Comment (RoiEXLab) [reply game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java]: Nice, I hope the other usage is guaranteed to be non-null otherwise that's going to be a problem (for obvious reasons the other usage you mentioned is not in the diff so I can't tell)

[the remark this answers is not in the document]
Comment (RoiEXLab) [reply game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java]: Feel free to merge if the other usage is safe 👍

[the remark this answers is not in the document]
Comment (asvitkine) [reply game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java]: Yep, the other usage was non-null.
